yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -150,16 +138,19 @@ export class Archiver implements L2BlockSource, UnverifiedDataSource, ContractDa
     this.log(`Retrieved ${retrievedBlocks.retrievedData.length} block(s) from chain`);
 
     // store retrieved rollup blocks
-    this.l2Blocks.push(...retrievedBlocks.retrievedData);
+    await this.store.addL2Blocks(retrievedBlocks.retrievedData);
```

Comment:
You will need to write the L2 blocks last. Otherwise the sequencer will be subject to race conditions on the availability of contract data bytecode